db/schema.sql

```diff
@@ -202,7 +200,7 @@ CREATE TABLE aggregate_share_jobs(
     CONSTRAINT unique_aggregate_share_job_task_id_interval_aggregation_param UNIQUE(task_id, batch_interval, aggregation_param),
     CONSTRAINT fk_task_id FOREIGN KEY(task_id) REFERENCES tasks(id)
 );
-CREATE INDEX aggregate_share_jobs_interval_containment_index ON aggregate_share_jobs USING gist (task_id, batch_interval);
+CREATE INDEX aggregate_share_jobs_interval_containment_index ON aggregate_share_jobs USING gist (batch_interval);
```

Comment:
To make sure I understand this change:
* The planner already wasn't using this index in relevant queries (i.e. queries finding collect jobs/aggregate share jobs by time or interval).
* With this change, the index is used.
* But the index no longer covers task_id, so now we will have to sort through collect jobs/aggregate share jobs from all tasks in order to find the ones we care about.

(correct me if I'm wrong on any of this, please)

If so, I think we can make this change, but this won't scale too well as the number of tasks increases. We may need to do something eventually, but for now this is a good improvement.

Reply:
Slight correction: The planner was already using this index, but it was only ever putting the `batch_interval` column in the index condition when it did so, not the `task_id`. The `task_id` condition from the query gets put in the "filter" part of the index scan both before and after this index change. I agree regarding the concern at large numbers of tasks. Bitmap index scans are pretty good, but not optimal.